Automattic/wp-calypso
# GDPR banner: render on server

This change attempts to render the GDPR banner on the server side, to reduce the time it takes to be displayed. It takes up a significant amount of space in mobile layouts and it becomes the largest element to be painted, affecting LCP scores. By rendering it earlier, we can make sure it doesn't unduly penalise LCP scores for SSRed, logged-out pages like `/log-in`.

In order to SSR this, we must look at the request cookies, so a mechanism was added to make these available to the layout, in order to have them passed as props to components that need them.

#### Changes proposed in this Pull Request

* Add mechanism to pass  boolean to layout, indicating whether or not to display the banner
* Write country code cookie if not present in initial request but known from Geo IP header
* Refactor analytics utils to add functionality and split existing methods
* Modify GDPR banner component to render on the server, instead of only on the client
* Modify GDPR banner to attempt to fetch country code if unknown at first
* Remove animation when the GDPR banner gets displayed. This delays LCP, and was getting triggered a second time when the page hydrates.

#### Testing instructions

* Modify the config for your test environment to ensure that `gdpr-banner` is set to `true`
* Load `/log-in` as a logged-out user with no cookies
* Ensure the cookie bar gets displayed as part of the first render
* Click the Accept button
* Ensure the cookie bar animates away
* Reload the page
* Ensure the cookie bar does not get displayed as part of the first render
